### DISCUSSION TITLE
✅ 🔡 i18n: Up-to-date source strings for Elementor Pro not available in Glotpress

### DISCUSSION THREAD

**Discussion by toineenzo** [elementor/elementor#25373] [Feature Request]
### Edit 2: This is now fixed
Glotpress updated and strings should be kept up-to-date and in sync with the Elementor Pro source strings. 

### Edit 1
This request was initially made to contribute Dutch translations for Elementor Pro. Currently, this discussion addresses the issue that the source strings available on translate.elementor.com (GlotPress) are not aligned or up-to-date with the source strings available in the Elementor Pro zip file (i.e., the .POT file). This issue should be resolved by updating the GlotPress installation and ensuring the pipeline functions correctly. This includes automatically importing the source strings/.POT files into GlotPress with every release or commit to Elementor Pro. This ensures accessibility and better understanding for people who can't read / understand English. 

### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

The strings for Elementor Pro are currently not up-to-date for the Dutch language, that's why I updated all the strings and added the missing ones. So here is my contribution, as previously discussed with @nicholaszein.

Please see the attached .po and .mo files and the used .pot template file at the link below. I checked all the strings inside Elementor Pro (all functions and widgets) and everything is looking good :). Also the Dutch glossary has been used and checked with the strings for consistency, although some spaces have been used for better readability. The .pot template file used, is the latest one available for Elementor Pro 3.20 (stable).

ps. There are still some strings though that are not translatable or not working, like everything inside the Notes feature, even though those strings are included inside my file. I'll try to post some screenshots in this request.
pps. I still have a lot of strings on https://translate.elementor.com/projects/elementor-pro/nl/default/ that have the 'waiting' status, but those are out-of-date right now. So please keep that in mind if/when the local glotpress system will be used again in the future.

Thank you!
-Toine

**These strings will now all be submitted and approved by me on translate.elementor.com and eventually pushed to production via an in-WordPress translation update. (so don't submit or manually use these files again, these are available here strictly for archive purposes only.)**
[Dutch strings for Elementor Pro 3.21-beta3 - Download](https://github.com/elementor/elementor/files/14843651/Elementor.Pro.3.21.0-beta1.-.nl_NL.zip)
[Dutch strings for Elementor Pro <=3.20](https://github.com/elementor/elementor/files/14502960/Elementor.Pro.nl_NL.translation.strings.zip)

### Describe the solution you'd like.

n/a

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by toineenzo** [comment]
@nicholaszein any news on when the translation will be implemented? :)

**Reply by nicholaszein** [reply]
@toineenzo I'll have a answer to you by Tuesday! 😉

**Reply by toineenzo** [reply]
> @toineenzo I'll have a answer to you by Tuesday! 😉

Any news? @nicholaszein

**Comment by toineenzo** [comment]
Strings updated to Elementor Pro 3.21-beta1

**Comment by nicholaszein** [comment]
Hey @toineenzo!

Apologies for the delay. Could you please indicate what is your username in https://translate.elementor.com/projects/elementor-pro?

I'm trying my best to push your translations to production before the release of Pro 3.21.

Thanks 🙏

**Reply by toineenzo** [reply]
@nicholaszein 
Alright, my username on there is user-5fa0d0b7ead63. If you can change this to toineenzo, that would be nice. 
Please note that all the translations that are currently in waiting status for Dutch there, are all outdated.
So please don't approve those strings and reimport the .po file you can download on this GitHub page above! ;). It may also be good to check, the strings on* translate.elementor.com are up-to-date with the strings in Elementor Pro 3.20/3.21 so perhaps the .pot file needs to be updated for all languages?
Or you could change my user permissions / make me a translation validator so I can do this myself and/or accept Dutch translations in the future. 

Thanks
Toine

**Comment by rami-elementor** [comment]
@toineenzo Thank you for translating Elementor.

I approve over 1100 strings. Will try to give you admin permissions for your language.

**Reply by toineenzo** [reply]
Thank you @rami-elementor, please note that all the strings that are in waiting status, are outdated. So I hope that the import option will be visible when you set my account to admin so I can import the up-to-date .po file I posted here.

**Reply by nicholaszein** [reply]
Hi @toineenzo!

It should appear by now. Please check it out and let me know! 

Thank you for being so patient! 🙏

**Reply by toineenzo** [reply]
@nicholaszein I DO see the import option now, and I successfully imported my file, but I get a critical error when approving strings in bulk :( (so i have to approve them one by one)
<img width="842" alt="image" src="https://github.com/elementor/elementor/assets/4921397/2bfbd7b0-3fff-4e34-8d91-3bac0f0eb8af">
<img width="779" alt="image" src="https://github.com/elementor/elementor/assets/4921397/cd8f6487-5040-4d21-aaf8-7afb97eb2625">
And to manually approve every string one by one, also triggers the Cloudflare protection every minute.
Lastly, are the source strings up-to-date in translate.elementor.com? The .pot file I have right now of 3.21.0-beta3 contains 2431* strings but translate.elementor.com contains 2510 strings.

**Reply by nicholaszein** [reply]
@toineenzo, this is @rami-elementor's territory, and I'll see what I can do about it. Sorry for the inconvenience!

**Comment by toineenzo** [comment]
<img width="396" alt="image" src="https://github.com/elementor/elementor/assets/4921397/245db782-7b2f-4631-9975-0c6e30e0e832">
It's correctly pushing to production though :)

**Reply by nicholaszein** [reply]
🙌

**Comment by toineenzo** [comment]
@nicholaszein @rami-elementor
Nevermind, i'll just have a macro running this night that's approving the strings one by one with a automatic keystroke, so everything will be 100% approved and pushed to production by tomorrow ;-) A little bit of a dirty trick, but probably faster than wating for a fix for the bulk approve critical error 🤷😁. Updating to GlotPress 4.0+ could fix this.
Also Cloudflare seems to be set up a bit too aggressive with the Rate limiting, but this probably done to secure us from vandals ;-)

Just to give this ticket already some closure: I'll try to keep the Dutch translation up-to-date every beta release, so the strings will be up-to-date before every version release (or approve any submitted strings). Just keep in mind to keep the source strings available in GlotPress up-to-date with the strings in Elementor, so everything will be translatable before release. All in the name of accessibility 🥰

Have a good day and i'll be looking forward to the upcoming Elementor Event and future!

**Reply by nicholaszein** [reply]
@toineenzo thank you so much for the update! We're definitely going to consider your advices!

**Comment by toineenzo** [comment]
I'll close this issue for now. I'll be creating a bug report soon about some areas that don't appear to be working with the translations (Like everything inside the Notes feature), even though the strings are included in the .po / .pot file.

**Comment by toineenzo** [comment]
Hi @nicholaszein @rami-elementor , could you take a look again at the new strings that are included with the new 3.21 release? It doesn't seem translate.elementor.com contains all the new strings and is being synced with the source.
Thanks!

**Reply by nicholaszein** [reply]
Hey @toineenzo, I'm aware of what you're talking about, however we're going to have to wait a little, as there are some holidays coming, as well as we need some help from some people to figure out the problems you reported with the translate site (and they are on holiday). Sorry about that, it's a team effort rather than a one man show, and sometimes we're not all in sync. Please, bear with us a little! 🙏

**Reply by toineenzo** [reply]
@nicholaszein No problem! Thank you for your fast response. I'll keep this issue open for now then, to communicate :)

**Reply by nicholaszein** [reply]
Hey @toineenzo!

Just to let you know, we are currently working on some issues related to Elementor Pro translations, and we haven't yet had the opportunity to update the new strings because of this.

We haven't forgot you, and we wanted to thank you for your contribution and patience. We know it's been a while. So thank you! 🙏 

We'll update you when we have news! 😉 

Kind regards

**Reply by toineenzo** [reply]
Hi @nicholaszein, thanks for your message. I'm hoping the 'translation situation' will be improved soon, and all strings will be translatable.

**Comment by toineenzo** [comment]
Hi @nicholaszein @rami-elementor , hope you're doing great. I was curious when the translation situation will be addressed, it's especially important now that a lot of the source strings have been changed. Thanks!
Ps I do understand there are some more important things to fix right now ;) but maybe some time indication would be nice.

**Comment by toineenzo** [comment]
@nicholaszein @rami-elementor Hi, I just saw that the Glotpress instance got updated. This fixed the fatal error when bulk approving strings.
I also see now that new strings are added! Thanks for (finally) fixing it and I will soon update my translations. 
Toine
PS can my Glotpress username maybe be changed from user-5fa0d0b7ead63 to toineenzo?
![image](https://github.com/elementor/elementor/assets/4921397/0e7cc10a-d371-4924-958b-2f6bafd06ee9)

**Reply by toineenzo** [reply]
Forgot to add: is it also possible to add the Dutch formal language type? (NL_nl_formal Wordpress language code)
Edit: looks like it's not supported yet in Glotpress https://github.com/GlotPress/GlotPress/issues/1460

**Reply by nicholaszein** [reply]
Hi @toineenzo!

Yes, we did a bunch of updates to our GlotPress platform last week (did you see the AI suggestions?), and the issues you've reported should all be fixed. Please, let us know if you find any problems. We are still investigating problems with some languages, we want to make sure yours is not one of them.

As for the formal Dutch, you are correct. Unfortunately, it's not supported yet.

I will see if your username can be changed. It would actually be a "nice to have" feature, however due to how we handle subscriptions to the platform I believe we cannot change it at this point.

Thank you for being so patient with us. We had a lot to work on, and it took more time than we wished it would take. 🙏 

Kind regards 🙂

**Reply by toineenzo** [reply]
Thank you, @nicholaszein :)!
I already did some translations and everything looks/works good! Also getting correctly pushed to production.
Yes, I saw the OpenAI feature, that's nice! I will try it out., but I'm more of a DeepL person though. ;)

ps Maybe a good feature to have, to combat bots/malicious translations, is to make it a requirement to have an active Elementor Pro license when logging in with your Elementor account on translate.elementor.com

I'll close this issue for now, if there's anything in the future, i'll create a new one.
Thanks and have a good one.

**Reply by nicholaszein** [reply]
Thanks for the feedback @toineenzo!

Yes, we are considering including DeepL integration. I'll send this feedback to the team! 😉

As for the suggestion, that's actually a great suggestion! Simple and effective. This is definitely going to the team! 💪

Thank you for contributing! 🙏

You rock! 👊🤘

**Reply by toineenzo** [reply]
@nicholaszein i encountered one error still: when approving strings in bulk (selecting multiple and choosing Approve) still gives a Fatal error

**Comment by toineenzo** [comment]
@nicholaszein I think there is a typo in the code for this string
![image](https://github.com/user-attachments/assets/fb4eb850-8774-46e8-8a49-7e36374e2c6a)
elementor-pro/assets/js/form-submission-admin.js#L3152

I also noticed that the total strings available went from 2722 to 254. I'm not sure if that should be like that?

**Reply by toineenzo** [reply]
@rami-elementor maybe you can take a look at this. It looks like most of the source strings have disappeared from Glotpress.

**Comment by toineenzo** [comment]
<img width="1085" alt="afbeelding" src="https://github.com/user-attachments/assets/fe82344b-1417-4c5e-b767-e29d6c70d528">
Looks like its fixed now, so ill close this issue again @nicholaszein @rami-elementor
